ipatests/test_integration/test_adtrust_install.py

```diff
@@ -272,3 +272,43 @@ def test_ipa_user_s4u2self_pac(self):
         finally:
             tasks.kinit_admin(self.master)
             self.master.run_command(['ipa', 'user-del', user])
+
+    def test_adtrust_agents_are_recreated_after_upgrade(self):
+        """Test if adtrust agents, which are removed from LDAP prior to
+        an upgrade, are recreated after the upgrade
+        Related: https://pagure.io/freeipa/issue/8543"""
+        passwd = self.master.config.dirman_password
+        host = self.replicas[0]
+        self.unconfigure_replica_as_agent(host)
+        res = self.master.run_command(['ipa-adtrust-install',
+                                       '--add-agents', '--add-sids',
+                                       '-a', passwd, '-U'])
+        assert "Setup complete" in res.stdout_text
+
+        search_trust_agents = textwrap.dedent("""
+        cn=adtrust agents,cn=sysaccounts,cn=etc,{base_dn}
+        """.format(base_dn=str(host.domain.basedn)).strip())
+        tasks.ldapsearch_dm(host, search_trust_agents, "", ok_returncode=[0])
+        remove_trust_agents = textwrap.dedent("""
+             dn: cn=adtrust agents,cn=sysaccounts,cn=etc,{base_dn}
+             changetype: delete
+             EOF
+             """.format(base_dn=str(host.domain.basedn)).strip())
```

Comment:
Could you please explain why you should delete `cn=adtrust agents` DN? This DN expected to exist on all IPA replicas because it is part of our configuration and is replicated to all trees. Why should it be deleted in the test?

Reply:
IfI understood that correctly, the original bugzilla implies that adtrust agents should be recreated with upgrade after they were removed. Hence the removal.

Reply:
The bugzilla https://bugzilla.redhat.com/show_bug.cgi?id=1778777 describes a different issue: the entry 'cn=adtrust agents,cn=sysaccounts,cn=etc,$BASEDN still exists but is missing some *members*. In order to reproduce the issue, you need to delete some attribute values from the entry, not the whole entry.
The fix re-adds member: fqdn=hostname,cn=computers,cn=accounts,$BASE for each member: krbbprincipalname=cifs/hostname@realm found in the entry.